aml/org.testeditor.aml.dsl/src/org/testeditor/aml/dsl/naming/AmlQualifiedNameProvider.xtend

```diff
@@ -14,19 +14,27 @@ package org.testeditor.aml.dsl.naming
 
 import javax.inject.Inject
 import org.eclipse.emf.ecore.EObject
-import org.eclipse.xtext.naming.IQualifiedNameConverter
 import org.eclipse.xtext.xbase.scoping.XbaseQualifiedNameProvider
 import org.testeditor.aml.AmlModel
+import org.testeditor.dsl.common.util.classpath.ClasspathUtil
 
 class AmlQualifiedNameProvider extends XbaseQualifiedNameProvider {
 
-	@Inject
-	extension IQualifiedNameConverter
+	@Inject ClasspathUtil classpathUtil
 
 	override getFullyQualifiedName(EObject obj) {
 		val result = switch (obj) {
-			AmlModel:
-				obj.package.toQualifiedName
+			AmlModel: {
+				if (obj.package === null) {
+					val package = classpathUtil.inferPackage(obj)
+					if (!package.nullOrEmpty) {
+						obj.package = package
```

Comment:
Is this the return value as well? Unclear to me.
It also looks a bit unclean to me that the underlying model is modified when the name provider is asked for the qualified name of it. A better solution for caching the calculated value would be to use an eAdapter.